### PR TITLE
Fixes runtimes from firing projectiles, adds unit test for firing guns

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
#57214 fixed a runtime for ninja gloves by changing a handful of procs related to projectiles, including preparePixelProjectile(), to expect modifiers instead of params. It did not do the same for all of the other places that called preparePixelProjectile, including the proc that makes ammo casings fire, making every gunshot runtime (twice). This PR makes everywhere that calls preparePixelProjectile with param info translate it to modifiers first instead.

This PR also adds a unit test to ensure that a basic ballistic pistol is capable of spawning (loaded and chambered), and being able to fire and hit a target for the correct damage amount.

## Why It's Good For The Game
Guns no longer spam runtime logs, prevents this from happening in the future

## Changelog
:cl: Ryll/Shaps
fix: Firing guns no longer runtimes
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -850,6 +850,9 @@
 
 //Spread is FORCED!
 /obj/projectile/proc/preparePixelProjectile(atom/target, atom/source, modifiers, spread = 0)
+	if(modifiers && !islist(modifiers))
```
Comment (Mothblocks) [inline code/modules/projectiles/projectile.dm:853]:
```suggestion
	if(!isnull(modifiers) && !islist(modifiers))
```

I know it's unlikely, but while you're doing a test like this you might as well cover your bases and don't let FALSE/0 through. Plus, it's more explicit.

File: code/modules/unit_tests/projectiles.dm [hunk 2]
```diff
@@ -3,3 +3,27 @@
...
 		if(initial(projectile.movement_type) & PHASING)
 			Fail("[path] has default movement type PHASING. Piercing projectiles should be done using the projectile piercing system, not movement_types!")
+
+/datum/unit_test/gun_go_bang/Run()
+	// test is for a ballistic gun that starts loaded + chambered
+	var/obj/item/gun/test_gun = allocate(/obj/item/gun/ballistic/automatic/pistol)
+	var/mob/living/carbon/human/victim = allocate(/mob/living/carbon/human)
+	var/mob/living/carbon/human/gunner = allocate(/mob/living/carbon/human)
+
+	ADD_TRAIT(victim, TRAIT_RESISTLOWPRESSURE, INNATE_TRAIT) // so pressure damage doesn't throw off our damage check
```
Comment (Mothblocks) [inline code/modules/unit_tests/projectiles.dm:13]: Man, I really need to make the testing place not in space by default.

File: code/modules/unit_tests/projectiles.dm [hunk 3]
```diff
@@ -3,3 +3,27 @@
...
+	var/obj/projectile/loaded_bullet = loaded_casing.loaded_projectile
+	TEST_ASSERT(loaded_bullet, "Ammo casing has no loaded bullet")
+
+	gunner.put_in_hands(test_gun, forced=TRUE)
+	gunner.set_combat_mode(FALSE) // just to make sure we know we're not trying to pistol-whip them
+	var/expected_damage = loaded_bullet.damage
+	loaded_bullet.def_zone = BODY_ZONE_CHEST
+	var/did_we_shoot = test_gun.afterattack(victim, gunner)
+	TEST_ASSERT(did_we_shoot, "Gun does not appeared to have successfully fired.")
+	TEST_ASSERT(victim.getBruteLoss() == expected_damage, "Victim took incorrect amount of damage, expected [expected_damage], got [victim.getBruteLoss()].")
```
Comment (Mothblocks) [inline code/modules/unit_tests/projectiles.dm:26]: Use `TEST_ASSERT_EQUAL`.

File: code/modules/unit_tests/projectiles.dm [hunk 4]
```diff
@@ -3,3 +3,27 @@
...
+
+	var/obj/item/bodypart/expected_part = victim.get_bodypart(BODY_ZONE_CHEST)
+	TEST_ASSERT(expected_part.brute_dam == expected_damage, "Intended bodypart took incorrect amount of damage, either it hit another bodypart or armor was incorrectly applied. Expected [expected_damage], got [expected_part.brute_dam].")
```
Comment (Mothblocks) [inline code/modules/unit_tests/projectiles.dm:29]: Use `TEST_ASSERT_EQUAL`.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Thanks for the tests!

File: code/modules/unit_tests/projectiles.dm (hunk 2 above)
Comment (Ryll-Ryll) [reply code/modules/unit_tests/projectiles.dm:13]: YEAH, wasted 30 minutes figuring out where that extra 2.4 brute was coming from

Comment (Mothblocks) [reply code/modules/unit_tests/projectiles.dm:13]: :sob:

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**